sass/components/forms/_switches.scss

```diff
@@ -56,36 +56,40 @@
     transition: left 0.3s ease, background .3s ease, box-shadow 0.1s ease, transform .1s ease;
   }
 
-  &:before {
-    background-color: transparentize($switch-bg-color, .85);
-  }
-
   &:after {
-    background-color: $switch-unchecked-bg;
+    background-color: $switch-thumb-unchecked-color;
     box-shadow: 0px 3px 1px -2px rgba(0, 0, 0, 0.2), 0px 2px 2px 0px rgba(0, 0, 0, 0.14), 0px 1px 5px 0px rgba(0, 0, 0, 0.12);
   }
 }
 
+input[type=checkbox]:not(:disabled) ~ .lever:active:before,
+input[type=checkbox]:not(:disabled).tabbed:focus ~ .lever::before,
+input[type=checkbox]:not(:disabled) ~ .lever:hover::before {
+    transform: scale(2.4);
+}
+
+input[type=checkbox]:checked:not(:disabled) ~ .lever:hover::before {
+  background-color: $switch-reaction-checked-hover-color;
+}
+
 // Switch active style
 input[type=checkbox]:checked:not(:disabled) ~ .lever:active::before,
 input[type=checkbox]:checked:not(:disabled).tabbed:focus ~ .lever::before {
-  transform: scale(2.4);
-  background-color: transparentize($switch-bg-color, .85);
+  background-color: $switch-reaction-checked-focus-color;
+}
+
+input[type=checkbox]:not(:disabled) ~ .lever:hover::before {
+  background-color: $switch-reaction-unchecked-hover-color;
 }
 
 input[type=checkbox]:not(:disabled) ~ .lever:active:before,
 input[type=checkbox]:not(:disabled).tabbed:focus ~ .lever::before {
-  transform: scale(2.4);
-  background-color: rgba(0,0,0,.08);
+  background-color: $switch-reaction-unchecked-focus-color;
 }
 
 // Disabled Styles
 .switch input[type=checkbox][disabled] + .lever {
   cursor: default;
-  background-color: rgba(0,0,0,.12);
+  opacity: 0.5;
```

Comment:
Why did you change this bg-color ?  opacity: .5 isn't a good contrast

Reply:
I made it according to material design guidelines:
https://material.io/components/switches#theming
Do you think we should keep existing behavior?

I chose `opacity` over explicit `background-color` to work better with custom component styling. Not a very big thing though, I could also use background-color with a couple of tweaks

Reply:
Current materializecss behaviour (upper enabled, lower disabled):
ON
![image](https://user-images.githubusercontent.com/1275813/115677185-880a0000-a350-11eb-9f24-20f95608f06f.png)
OFF
![image](https://user-images.githubusercontent.com/1275813/115677289-a07a1a80-a350-11eb-95d4-0eef968e9912.png)

My implementation: 
![image](https://user-images.githubusercontent.com/1275813/115677352-af60cd00-a350-11eb-88ab-cadcb7c70cde.png)
![image](https://user-images.githubusercontent.com/1275813/115677417-be477f80-a350-11eb-9f14-9dc9cb3c73ba.png)

Reply:
could you pleaese remove the "0 " on opactity: 0.5 ?

Reply:
Hi @roiLeo sure, I can do it for all my changes. 

But there is a small _but_: I checked scss folder for `0.x` and `.x` values: 86 occurrences vs 134
It seems there is no globally agreed guideline. 
And according to scss guidelines (https://sass-guidelin.es/), leading zeroes should be used. 

I don't care much which approach to use, but probably we should add a eslint rule to have a common style, what do you think? https://stylelint.io/user-guide/rules/number-leading-zero

Reply:
I didn't knew about this sass guideline maybe we should use leading zeroes for more readibility and make it required in .eslintrc

Reply:
I would create a separate PR with this change then, to avoid the unnecessary intersection between changes and also to hear the opinion of other members.
I hope it's ok

Reply:
Perfect @bugy. Let's do it like you wrote.